assets/app/view/game/part/track_node_path.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+require 'view/game/part/base'
+
+module View
+  module Game
+    module Part
+      class TrackNodePath < Base
+        needs :tile
+        needs :path
+        needs :color, default: 'black'
+        needs :width, default: 8
+        needs :dash, default: '0'
+
+        EDGE_PERP_ANGLES = [90, 30, -30, -90, -150, 150].freeze
+
+        EDGE_REGIONS = {
+          0 => TRACK_TO_EDGE_0,
+          1 => TRACK_TO_EDGE_1,
+          2 => TRACK_TO_EDGE_2,
+          3 => TRACK_TO_EDGE_3,
+          4 => TRACK_TO_EDGE_4,
+          5 => TRACK_TO_EDGE_5,
+        }.freeze
+
+        def self.lookup_pos_x(edge, distance)
```

Comment:
these don't need to be class methods since you're using @@

it should just be lookup_pos, then the next calls won't need to call TrackNodePath